### ISSUE TITLE
ESM Jest type export issue

### ISSUE THREAD

**Issue by jsagethuboo** [knex/knex#5548]
Knex version: 2.3.0
OS: Mac Ventura 13.2.1

I have a typescript project which is being used as ESM.  
When running tests with jest I have ts-jest set up to utilise ESM transforming (jest.config below)

When running any test it fails with the following message: "The requested module 'knex' does not provide an export named 'Knex'" 

Within the file in question this is the syntax used to import the type def `import { Knex } from 'knex';` 

I understand that esm seems to struggle with named exports from CJS modules so have also tried `import knex from 'knex'` then trying to reference knex.Knex which does not exist

```
{
  preset: 'ts-jest/presets/default-esm',
  testEnvironment: 'node',
  clearMocks: true,
  collectCoverage: false,
  coverageDirectory: 'coverage',
  workerIdleMemoryLimit: '512MB',
  collectCoverageFrom: ['./src/**/*', '!./src/**/*.test.ts'],
  coveragePathIgnorePatterns: [],
  coverageProvider: 'v8',
  coverageReporters: ['json', 'lcov', 'text', 'clover', 'cobertura'],
  setupFiles: ['<rootDir>jest.bootstrap.ts'],
  testPathIgnorePatterns: ['/node_modules/', '/.build/', '/config/'],
  transform: {
    '.(ts|tsx)': ['ts-jest', { useESM: true }],
  },
  moduleNameMapper: {
    '@/(.*)': '<rootDir>/src/$1',
    '@tests/(.*)': '<rootDir>/tests/$1',
  },
  transformIgnorePatterns: [`/node_modules/`, '\\.pnp\\.[^\\/]+$'],
  verbose: true,
  extensionsToTreatAsEsm: ['.ts'],
};
```

Any help/support would be appreciated

**Comment by FelipeEmerimRandon** [comment]
I also had this, sadly the only way around it I found is using @ts-ignore just before the import.

**Comment by mercmobily** [comment]
Closing this in favor of #5800 which addresses the root cause - Knex's need for proper ESM support through migration to modern  tooling (tsx/Vitest).

This issue experiences the same "The requested module 'knex' does not provide an export named 'Knex'" error as #6218 (Node 24 CLI  context), just manifesting in Jest/TypeScript tests. Both are symptoms of the broader ESM/CJS interoperability issues that #5800 will resolve.

  **Workarounds until #5800 is complete:**
  1. Use `@ts-ignore` before the import statement
  2. Or try `NODE_OPTIONS='--import tsx/esm'` when running tests

 Consolidating these ESM-related issues into #5800 for cleaner tracking. Once the migration to tsx/Vitest is complete, these ESM export issues should be resolved.

Both comments explain the consolidation rationale, provide workarounds, and reference the relationship between the issues.
